src/value.cpp

```diff
@@ -1045,17 +1023,23 @@ namespace Exiv2 {
     int TimeValue::read(const byte* buf, long len, ByteOrder /*byteOrder*/)
     {
         // Make the buffer a 0 terminated C-string for scanTime[36]
-        char b[] = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
-        std::memcpy(b, reinterpret_cast<const char*>(buf), (len < 12 ? len : 11));
+        char b[] = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
```

Comment:
```suggestion
        char b[15] = {};
```

Reply:
I have replaced that code by the usage of regular expressions. I added few more unit tests to be sure we are supporting the different formats mentioned in the ISO.